pkg/api/controller.go

```diff
@@ -1626,23 +1626,36 @@ func (c *Controller) validateStorageNamespace(storageNamespace string) error {
 }
 
 func (c *Controller) ensureStorageNamespace(ctx context.Context, storageNamespace string) error {
-	const (
-		dummyKey  = "dummy"
-		dummyData = "this is dummy data - created by lakeFS in order to check accessibility"
-	)
+	dummyKey := fmt.Sprintf("%s/dummy", c.Config.Committed.BlockStoragePrefix)
+	const dummyData = "this is dummy data - created by lakeFS in order to check accessibility"
 
-	obj := block.ObjectPointer{
+	// check if dummy file exists in previous lakeFS versions at the root of the storage namespace
+	objLegacyLocation := block.ObjectPointer{
 		StorageNamespace: storageNamespace,
 		IdentifierType:   block.IdentifierTypeRelative,
-		Identifier:       dummyKey,
+		Identifier:       "dummy",
 	}
 	objLen := int64(len(dummyData))
-	if _, err := c.BlockAdapter.Get(ctx, obj, objLen); err == nil {
-		return fmt.Errorf("found lakeFS objects in the storage namespace(%s): %w",
+
+	if _, err := c.BlockAdapter.Get(ctx, objLegacyLocation, objLen); err == nil {
+		return fmt.Errorf("found lakeFS objects in the legacy storage namespace(%s): %w",
 			storageNamespace, ErrStorageNamespaceInUse)
 	} else if !errors.Is(err, block.ErrDataNotFound) {
 		return err
 	}
+	// check if the dummy file exist
+	obj := block.ObjectPointer{
+		StorageNamespace: storageNamespace,
+		IdentifierType:   block.IdentifierTypeRelative,
+		Identifier:       dummyKey,
+	}
+
+	if _, err := c.BlockAdapter.Get(ctx, objLegacyLocation, objLen); err == nil {
```

Comment:
Bug?
```suggestion
	if _, err := c.BlockAdapter.Get(ctx, obj, objLen); err == nil {
```

Reply:
Probably merits a test in controller_test.go.  The important part is to test that re-creating another repo at the same location fails; less important is to verify that creating another repo at the same location as a "legacy" dummy repo also fails.

Reply:
isn't it what [use same storage namespace twice](https://github.com/treeverse/lakeFS/blob/master/pkg/api/controller_test.go#L203) test does?